models/topics.model.js

```diff
@@ -26,9 +26,31 @@ exports.fetchArticleById = async (article_id) => {
     return result.rows[0];
   }
 };
-exports.fetchArticles = async () => {
-  const queryStr =
-    "SELECT articles.article_id, articles.title,articles.topic, articles.author, articles.created_at, articles.votes, articles.article_img_url, CAST(COUNT(comments.body) AS INTEGER) AS comment_count FROM articles LEFT JOIN comments ON articles.article_id = comments.article_id GROUP BY articles.article_id,articles.title, articles.topic, articles.author, articles.created_at, articles.votes, articles.article_img_url ORDER BY articles.created_at DESC ";
+exports.fetchArticles = async (sort_by, order) => {
```

Comment:
If you default the parameters here using `sort_by = 'created_at'` for example, I think you would save yourself some of the if/else blocks down below. You should just be able to then check the validity of the sort/order and then reject if they don't match, otherwise use them in the query.